pkg/plugin/plugin_test.go

```diff
@@ -334,6 +420,47 @@ var _ = Describe("CNI Plugin", func() {
 				testAddDel(conf, false, true, "")
 			})
 		})
+		Context("with service VLAN and CVLAN", func() {
+			conf := fmt.Sprintf(`{
+				"cniVersion": "0.3.1",
+				"name": "mynet",
+				"type": "ovs",
+				"bridge": "%s",
+				"vlan": %d,
+				"vlanMode": "dot1q-tunnel",
+				"cvlan": [ {"id": 42} ]
+			}`, BRIDGE_NAME, VLAN_ID)
+			It("should successfully complete ADD and DEL commands", func() {
+				testQINQ(conf, true, "[42]")
+			})
+		})
+		Context("with service VLAN and range of CVLANs", func() {
+			conf := fmt.Sprintf(`{
+				"cniVersion": "0.3.1",
+				"name": "mynet",
+				"type": "ovs",
+				"bridge": "%s",
+				"vlan": %d,
+				"vlanMode": "dot1q-tunnel",
+				"cvlan": [ {"id": 24}, { "minID" : 42, "maxID" : 44 } ]
+			}`, BRIDGE_NAME, VLAN_ID)
+			It("should successfully complete ADD and DEL commands", func() {
+				testQINQ(conf, true, "[24, 42, 43, 44]")
+			})
+		})
+		Context("with service VLAN and no explicit CVLAN", func() {
+			conf := fmt.Sprintf(`{
+				"cniVersion": "0.3.1",
+				"name": "mynet",
+				"type": "ovs",
+				"bridge": "%s",
+				"vlan": %d,
+				"vlanMode": "dot1q-tunnel"
+			}`, BRIDGE_NAME, VLAN_ID)
+			It("should successfully complete ADD and DEL commands", func() {
+				testQINQ(conf, true, "[]")
+			})
```

Comment:
Please add tests to verify error handling:

* qinq is not enabled on the ovs db
* qinq is not enabled on the bridge
* cvlan is set but the vlanMode does not match

Reply:
> - qinq is not enabled on the ovs db

still TODO

> - qinq is not enabled on the bridge

As mentioned, I think there is nothing to enable on the bridge. Apart from configuring the port type, which is being tested.
> - cvlan is set but the vlanMode does not match

Added